stdlib/src/crypto/dsa/rpo_falcon512.rs

```diff
@@ -0,0 +1,149 @@
+use alloc::vec::Vec;
+
+use processor::{AdviceProvider, AdviceSource, ProcessState, handlers::EventError};
+use vm_core::{Felt, Word};
+
+/// Host handler function which pushes values onto the advice stack which are required for
+/// verification of a DSA in Miden VM.
+///
+/// Inputs:
+///   Operand stack: [PK, MSG, ...]
+///   Advice stack: \[ SIGNATURE \]
+///
+/// Outputs:
+///   Operand stack: [PK, MSG, ...]
+///   Advice stack: [...]
+///
+/// Where:
+/// - PK is the digest of an expanded public.
+/// - MSG is the digest of the message to be signed.
+/// - SIGNATURE is the signature being verified.
+///
+/// The advice provider is expected to contain the private key associated to the public key PK.
+///
+/// # TODO:
+/// Eventually, we might want to make this a handler stateful, containing the signing keys.
+/// These could be loaded into the advice provider by the host when loaded into the Host.
+pub fn falcon_sig_to_stack_handler(
+    advice_provider: &mut dyn AdviceProvider,
+    process: ProcessState,
+) -> Result<(), EventError> {
```

Comment:
I think this handler implementation should probably stay in the `processor`. I think of it as a "dummy" implementation that can be used with the default host, but is not suitable fore real-world uses.